services/todo-legacy/test/controller/todo-controller.test.ts

```diff
@@ -0,0 +1,121 @@
+import 'mocha';
+import * as _ from 'lodash';
+import { expect, sinon } from '@loopback/testlab';
+import { TodoController } from '../../controllers/todo-controller';
+import { TodoRepository } from '../../repositories/todo';
+
+describe('TodoController', () => {
+  let controller = new TodoController();
+  let fakeData = [
+      {
+        id: 1,
+        title: 'test1',
+        body: 'test1',
+      },
+      {
+        id: 2,
+        title: 'test2',
+        body: 'test2',
+      },
+  ];
+  describe('getTodo', () => {
+    it('returns all todos when called without ID', async () => {
+      let stub = sinon.createStubInstance(TodoRepository);
+      controller.repository = stub;
+      
+      stub.find.returns(fakeData);
+      let result = await controller.getTodo();
+      expect.equal(result, fakeData, 'data was correct');
+    });
+    it('returns the correct todo by ID', async () => {
+      let stub = sinon.createStubInstance(TodoRepository);
+      controller.repository = stub;
+      
+      stub.find.withArgs(1).returns(fakeData[0]);
+
+      let result = await controller.getTodoById(1);
+      expect.equal(result, fakeData[0], 'data was correct');
+    });
+    it('can filter by title', async () => {
+      let stub = sinon.createStubInstance(TodoRepository);
+      controller.repository = stub;
+
+      stub.find.withArgs(null, sinon.match({
+        where: sinon.match({ title: 'test2'}),
+      })).returns(fakeData[1]);
+      let result = await controller.getTodo('test2');
+      expect.equal(result, fakeData[1], 'data was correct');
+
+    });
```

Comment:
I originally wrote it this way because the controller was handling more of the control flow for repository calls.

Much of that logic migrated down a layer, so I can change the stubs.